rts/motoko-rts/src/memory/ic.rs

```diff
@@ -89,12 +93,19 @@ impl Memory for IcMemory {
     }
 }
 
-/// Page allocation. Ensures that the memory up to, but excluding, the given pointer is allocated.
+/// Page allocation. Ensures that the memory up to, and including, the given pointer is allocated,
+/// with the slight exception of not allocating the extra page for 0x100_000_000. For all
+/// `ptr > 0x100_000_000` the call to `wasm32::memory_grow` will trap.
+/// Invariant: ptr must be a reasonable pointer (i.e. less than 0x200_000_000).
 #[inline(never)]
 unsafe fn grow_memory(ptr: u64) {
-    debug_assert!(ptr <= 2 * u64::from(core::u32::MAX));
-    let page_size = u64::from(WASM_PAGE_SIZE.as_u32());
-    let total_pages_needed = ((ptr + page_size - 1) / page_size) as usize;
+    debug_assert!(ptr < 2 * u64::from(core::u32::MAX));
+    debug_assert_eq!(2 * u64::from(WASM_PAGE_SIZE.as_u32()) >> 16, 2);
```

Comment:
```suggestion
    debug_assert_eq!(u64::from(WASM_PAGE_SIZE.as_u32()) >> 16, 1);
```
Also  - why isn't this just a static assert - I don't think it depends on anything dynamic

Reply:
Probably.

Reply:
this code is different now